src/drivers/intel/dmic.c

```diff
@@ -579,9 +578,22 @@ static int select_mode(struct dai *dai,
 	else
 		mfir = modes->mfir_b;
 
-	mmin = find_min_int16(mfir, modes->num_of_modes);
-	count = find_equal_int16(idx, mfir, mmin, modes->num_of_modes, 0);
-	n = idx[count - 1];
+	/* Search fir_list[] decimation factors from start towards end. The found
+	 * last configuration entry with searched decimation factor will be used.
+	 */
+	count = 0;
```

Comment:
this is really an array index we are looking for ? I would rename this ```found```

Reply:
Yep, it's better.